soc/wch/ch32v/qingke_v4/Kconfig.defconfig

```diff
@@ -1,7 +1,7 @@
 # Copyright (c) 2025 MASSDRIVER EI (massdriver.space)
```

Comment:
TL;DR: I think we should rename this to ch32v20x_30x as microarchitecture != series, but ding this in a follow up PR is fine

Long story: I've been thinking about the same thing at https://github.com/nzmichaelh/zephyr/tree/ch32v006

The issue is that QingKe VXY isn't really a series as per https://docs.zephyrproject.org/latest/hardware/porting/soc_porting.html  The CH32V003 has a QingKe V2A, certain GPIO features, and AFIO mapping; while the CH32V006 and others in that series have a QingKe V2C and different GPIO, AFIO.

This is a coincidence, and as GPIO/AFIO configuration is not a feature of the microarchitecture, I don't think we should use the microarchitecture as the series.

I'd prefer keeping the directory as-is at the moment and revisiting in the future. Your choice though.

Reply:
It's only for sorting, there is no implication of having actual series behind it (at least for me): the support for Socs like CH592 was mentioned in the original discussion about this, and in the end, things are still set on soc/actual serie(ch32v30x, ch32v20x, etc) level.

Reply:
Alright we came up with something for blfb that might be better than this:
```
soc
  - bflb
    - bl60x
    - bl70x
    - bl70xl
    - bl61x
    - common
      - e24
      - e907
 ```
 so for wch it would be:
 ```
soc
  - ch32v
    - ch32v208
    - ch32v203
    - ch32v30x
    - ch32v003
    - common
      - qingke_v4b
      - qingke_v4c
      - qingke_v2a
 ```
What do you think?

Reply:
If we do organize the folders like this, would we keep the soc series (on the soc.yml and the Kconfig files) at Qingke_v4 ?

Reply:
> Alright we came up with something for blfb that might be better than this:
> 
> ```
> soc
>  - ch32v
>    - ch32v208
>    - ch32v203
>    - ch32v30x
>    - ch32v003
>    - common
>      - qingke_v4b
>      - qingke_v4c
>      - qingke_v2a
>      - qingke_v4f
> ```

Looks good. So ch32v is the family and ch32v208/203/30x/003 are the series. This fits https://docs.zephyrproject.org/latest/hardware/porting/board_porting.html and matches other socs like gd32 and rz.

This also addresses the problem on my ch32v006 branch - it's also a qingke_v2, except it's a qingke_v2c and has a different pinctrl setup than the ch32v003. Using this new layout I can add a `soc/ch32v/ch32v00x` series to cover the 002/005/006 etc.

Reply:
(I'm also happy if you want to split the soc rework from the CH32V208 specific changes)

Reply:
This is just discussion for now, that was the most relevant place accessible to discuss it. Maybe there should be a discussion thread or a discord channel for wch, i'd also rather split that from this, this should be pushed, ch32v303 and ch32v203 pushed, and then reorganize on that model once we discuss the details, we can discuss the detail in discussion / discord.